language/compiler/ir-to-bytecode/src/context.rs

```diff
@@ -335,6 +359,11 @@ impl<'a> Context<'a> {
         }
     }
 
+    /// Get the fake offset for the label. Labels will be fixed to real offsets after compilation
+    pub fn label_index(&mut self, label: Label) -> Result<CodeOffset> {
+        Ok(get_or_add_item(&mut self.labels, label)?)
```

Comment:
Okay, the whole label offset remapping process seems correct to me, but I found the usage of get_or_add_item confusing. `get_or_add_item` returns `TableIndex` and checks if the number of things in the table exceeds `TableIndex::max()`. However we're supposed to use `CodeOffset` to refer to locations in the vec of bytecodes. At least from my understanding, `TableIndex` and `CodeOffset` are different concepts even though they are both aliases of u16.

Reply:
These are fake offsets. The labels will be fixed to real offsets after the compilation is complete. 

Sure its just a coincidence that both are u16 right now, but we don't have any other tables of type `CodeOffset` right now. 
If these sizes switch, this function will not typecheck (see the return type), so I think its fine